authentik/flows/planner.py

```diff
@@ -229,18 +236,13 @@ def plan(self, request: HttpRequest, default_context: dict[str, Any] | None = No
             )
             context = default_context or {}
             # Bit of a workaround here, if there is a pending user set in the default context
-            # we use that user for our cache key
-            # to make sure they don't get the generic response
+            # we use that user for our cache key to make sure they don't get the generic response
             if context and PLAN_CONTEXT_PENDING_USER in context:
                 user = context[PLAN_CONTEXT_PENDING_USER]
             else:
                 user = request.user
-                # We only need to check the flow authentication if it's planned without a user
-                # in the context, as a user in the context can only be set via the explicit code API
-                # or if a flow is restarted due to `invalid_response_action` being set to
-                # `restart_with_context`, which can only happen if the user was already authorized
-                # to use the flow
-                context.update(self._check_authentication(request))
+
+            context.update(self._check_authentication(request, context))
```

Comment:
This change seems a bit scary but adding more `if`s seemed too convoluted. Performance loss seems negligible to me.

(If the `Outpost.objects.filter` call within the function is an issue, perhaps preface _that_ with an `if PLAN_CONTEXT_OUTPOST not in context`.)